src/components/menu-bar/menu-bar.jsx

```diff
@@ -20,91 +32,180 @@ import communityIcon from './icon--see-community.svg';
+                <div className={classNames(styles.menuBarItem)}>
                     <img
                         alt="Scratch"
                         className={styles.scratchLogo}
                         draggable={false}
                         src={scratchLogo}
                     />
                 </div>
-                <div className={classNames(styles.menuItem)}>
-                    <ComingSoonTooltip
-                        className={styles.comingSoon}
+                <div className={classNames(styles.menuBarItem, styles.hoverable)}>
+                    <MenuBarItemTooltip
+                        id="menubar-selector"
                         place="right"
-                        tooltipClassName={styles.comingSoonTooltip}
-                        tooltipId="menubar-selector"
                     >
                         <LanguageSelector />
-                    </ComingSoonTooltip>
+                    </MenuBarItemTooltip>
                 </div>
-                <div className={classNames(styles.menuItem)}>
-                    <ComingSoonTooltip
-                        className={styles.comingSoon}
-                        place="bottom"
-                        tooltipClassName={styles.comingSoonTooltip}
-                        tooltipId="file-menu"
+                <div
+                    className={classNames(styles.menuBarItem, styles.hoverable, {
+                        [styles.active]: props.fileMenuOpen
+                    })}
+                    onMouseUp={props.onClickFile}
+                >
+                    <div className={classNames(styles.fileMenu)}>File</div>
+                    <MenuBarMenu
+                        open={props.fileMenuOpen}
+                        onRequestClose={props.onRequestCloseFile}
                     >
-                        <div className={classNames(styles.fileMenu)}>File</div>
-                    </ComingSoonTooltip>
+                        <MenuItemTooltip id="new">
+                            <MenuItem>New</MenuItem>
+                        </MenuItemTooltip>
+                        <MenuSection>
+                            <MenuItemTooltip id="save">
+                                <MenuItem>Save now</MenuItem>
+                            </MenuItemTooltip>
+                            <MenuItemTooltip id="copy">
+                                <MenuItem>Save as a copy</MenuItem>
+                            </MenuItemTooltip>
+                        </MenuSection>
+                        <MenuSection>
+                            <LoadButton>{(renderFileInput, loadProject, loadProps) => (
+                                <MenuItem
+                                    onClick={loadProject}
+                                    {...loadProps}
+                                >
+                                    Upload from your computer
+                                    {renderFileInput()}
```

Comment:
This is cool! If I understand it correctly, you're passing in a function as a prop that renders the file input. This might be something I want to do for file and camera upload sounds/costumes/sprites...

Reply:
Yeah — it's an alternative to using a higher order component that eliminates the need for using an intermediary component. With a HOC this could have been passed in as a prop instead, but then you would need to create a component to wrap in the HOC. This way it can be inline.